Deprecate global_ordinals_hash and global_ordinals_low_cardinality

jimczi opened the pull request:
This change deprecates the `global_ordinals_hash` and `global_ordinals_low_cardinality` and
makes the `global_ordinals` execution hint choose internally if global ords should be remapped or use the segment ord directly.
These hints are too sensitive and expert to be exposed and we should be able to take the right decision internally based on the agg tree.

Relates #26014

jpountz commented on core/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/SignificantTermsAggregatorFactory.java:
```diff
@@ -291,44 +289,35 @@ Aggregator create(String name,
+                deprecationLogger.deprecated("global_ordinals_hash is deprecated. Please use [global_ordinals] instead.");
+                return GLOBAL_ORDINALS;
+            } else if ("map".equals(value)) {
+                return MAP;
```
let's do a switch/case?

jimczi commented:
> but on the other hand this PR moves part of the decision process from the factory to the aggregator, which feels wrong to me, and reduces testing coverage since we cannot force some execution modes anymore.

isn't it the opposite ? The factory now picks global ordinals execution with remapping or plain ordinals with the same heuristic than before. I removed the late decision in the aggregator and replaced it with https://github.com/elastic/elasticsearch/pull/26014. The aggregator is just able to remap or not based on the decision made in the factory.
For testing coverage I can add tests for the `GlobalOrdinalsStringTermsAggregator` which lacks testing anyway ?

jpountz commented:
Woops, I got a bit lost in the diff and misread your PR, sorry. This change is actually great the way it is. Thanks!

+1 to add tests to `GlobalOrdinalsStringTermsAggregator`.
